### PR TITLE
PWGHF: rapidity cut apply in Jpsi task

### PR DESCRIPTION
**Pull request by zhangbiao-phy**
Change eta cut with rapidity cut for Jpsi task

### CODE REVIEW

File: Analysis/Tasks/PWGHF/taskJpsi.cxx [hunk 1]
```diff
@@ -95,31 +94,35 @@ struct TaskJpsiMC {
     {{"hPtRecSig", "2-prong candidates (rec. matched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hPtRecBg", "2-prong candidates (rec. unmatched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      {"hPtGen", "2-prong candidates (gen. matched);#it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hPtGenSig", "2-prong candidates (matched);#it{p}_{T}^{gen.} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskJpsi.cxx]: `(matched)` -> `(rec. matched)`

File: Analysis/Tasks/PWGHF/taskJpsi.cxx [hunk 2]
```diff
@@ -95,31 +94,35 @@ struct TaskJpsiMC {
...
       }
-      if (cutEtaCandMax >= 0. && std::abs(candidate.eta()) > cutEtaCandMax) {
-        //Printf("MC Rec.: eta rejection: %g", candidate.eta());
+      if (cutYCandMax >= 0. && std::abs(YJpsi(candidate)) > cutYCandMax) {
+        //Printf("MC Rec.: Y rejection: %g", YJpsi(candidate));
         continue;
       }
       if (candidate.flagMCMatchRec() == 1 << JpsiToEE) {
-        registry.fill(HIST("hPtRecSig"), candidate.pt());
+        auto indexMother = RecoDecay::getMother(particlesMC, candidate.index0_as<aod::BigTracksMC>().label_as<soa::Join<aod::McParticles, aod::HfCandProng2MCGen>>(), 443, true);
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskJpsi.cxx]: Add documentation comment:
`// Get the corresponding MC particle.`

File: Analysis/Tasks/PWGHF/taskJpsi.cxx [hunk 3]
```diff
@@ -131,11 +134,11 @@ struct TaskJpsiMC {
     // MC gen.
     //Printf("MC Particles: %d", particlesMC.size());
     for (auto& particle : particlesMC) {
-      if (cutEtaCandMax >= 0. && std::abs(particle.eta()) > cutEtaCandMax) {
-        //Printf("MC Gen.: eta rejection: %g", particle.eta());
-        continue;
-      }
       if (particle.flagMCMatchGen() == 1 << JpsiToEE) {
+        if (cutYCandMax >= 0. && std::abs(RecoDecay::Y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode()))) > cutYCandMax) {
+          //Printf("MC Gen.: Y rejection: %g", RecoDecay::Y(array{particle.px(), particle.      py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode())));
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/taskJpsi.cxx]: `particle.      py()` -> `particle.py()`

**Review by vkucera — CHANGES_REQUESTED** [review]
Hi @zhangbiao-phy , some minor fixes. Otherwise looks ready to me.

**Comment by zhangbiao-phy** [comment]
> Hi @zhangbiao-phy , some minor fixes. Otherwise looks ready to me.

Hi @vkucera, Thanks a lot! I will fix it now.

**Review by vkucera — APPROVED** [review]
Thanks a lot, for the lightning-fast fix.
@ginnocen This looks ready to me.

**Review by ginnocen — APPROVED** [review]

**Pull request merged**